consul/install.sls

```diff
@@ -5,48 +5,44 @@
 # method is to download a zip, verify it, and extract consul into the system
 # path.
 
+include:
+  - .service
+
 download_package:
-  file.managed:
-    - name: /tmp/consul.zip
+  archive.extracted:
+    - name: /usr/bin/
     - source: {{ consul.consul_zip }}
     - source_hash: {{ consul.consul_zip_checksum }}
-
-unzip_package:
-  module.run:
-    - name: archive.unzip
-    - zip_file: /tmp/consul.zip
-    - dest: /tmp/consul
-    - require:
-      - file: download_package
-
-install_consul_bin:
-  file.copy:
-    - name: /usr/bin/consul
-    - source: /tmp/consul/consul
-    - require:
-      - module: unzip_package
+    - archive_format: zip
+    - if_missing: /usr/bin/consul
 
 permission_consul_bin:
   file.managed:
     - name: /usr/bin/consul
     - mode: 0755
+    - require:
+      - archive: download_package
 
 consul_data_directory:
   file.directory:
     - name: {{ consul.data_dir }}
     - makedirs: True
+    - require_in:
+      - service: start_consul_service
```

Comment:
This should probably be required in the `configure_consul_service` state instead of the service.